src/aws-codeartifact/domain.ts

```diff
@@ -0,0 +1,339 @@
+import { Annotations, ArnFormat, IResource, Lazy, Resource, Stack, Token, TokenComparison } from 'aws-cdk-lib';
+import { CfnDomain, CfnDomainProps } from 'aws-cdk-lib/aws-codeartifact';
+import {
+  AddToResourcePolicyResult,
+  Grant,
+  GrantWithResourceOptions,
+  IGrantable,
+  PolicyDocument,
+  PolicyStatement,
+  principalIsOwnedResource,
+} from 'aws-cdk-lib/aws-iam';
+import { IKey, Key } from 'aws-cdk-lib/aws-kms';
+import { Construct } from 'constructs';
+
+/**
+ * Represents a CodeArtifact Domain
+ */
+export interface IDomain extends IResource {
+  /**
+   * The ARN of the Domain
+   *
+   * @attribute
+   */
+  readonly domainArn: string;
+
+  /**
+   * The name of the Domain
+   *
+   * @attribute
+   */
+  readonly domainName: string;
+
+  /**
+   * The ARN of the key used to encrypt the Domain
+   *
+   * @attribute
+   */
+  readonly domainEncryptionKey?: string;
+
+  /**
+   * 12-digit account number of the AWS account that owns the domain that contains the Domain.
+   *
+   * @attribute
+   */
+  readonly domainOwner: string;
+
+  /**
+   * The KMS key used to encrypt the Domain
+   */
+  readonly encryptionKey?: IKey;
```

Comment:
If we don't automatically create the key, there would be no need to publish the domainEncryptionKey and the encryptionKey.

Reply:
The `domainEncryptionKey` may not an `@attribute`.

Also, if the `domainEncryptionKey` is to remain, it might be better named `domainEncryptionKeyArn`.

Reply:
As I understand, we need the `@attribute`, because `EncryptionKey` is an attribute of this resource (see [docs](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-codeartifact-domain.html#aws-resource-codeartifact-domain-return-values)).

![image](https://github.com/user-attachments/assets/4ff50987-987d-4140-9c3a-18d89f071fbb)

and according to https://github.com/aws/aws-cdk/blob/b49032b3a6e549783b45492ffc76880fbcd58e68/docs/DESIGN_GUIDELINES.md?plain=1#L684-L686

> AWS constructs must expose all resource attributes defined in the underlying
CloudFormation resource as readonly properties of the class
_[awslint:resource-attribute]_.

> All properties that represent resource attributes must include the JSDoc tag
**@attribute** _[awslint:attribute-tag]_.

> All attribute names must begin with the type name as a prefix
(e.g. ***bucket*Arn** instead of just **arn**) _[awslint:attribute-name]_. This
implies that if a property begins with the type name, it must have an
**@attribute** tag.

> All resource attributes must be represented as readonly properties of the
resource interface _[awslint:attribute-readonly]_.

Reply:
I see, nice catch! thanks for the clarification!